readthedocs/builds/views.py

```diff
@@ -54,14 +55,49 @@ def post(self, request, project_slug):
             return HttpResponseForbidden()
 
         version_slug = request.POST.get('version_slug')
-        version = get_object_or_404(
-            self._get_versions(project),
-            slug=version_slug,
-        )
+        build_pk = request.POST.get('build_pk')
+
+        if build_pk:
+            # Filter over external versions only when re-triggering a specific build
+            version = get_object_or_404(
+                Version.external.public(self.request.user),
+                slug=version_slug,
+                project=project,
+            )
+
+            build = get_object_or_404(
+                Build.objects.all(),
+                pk=build_pk,
+                version=version,
+            )
+            if build != Build.objects.filter(version=version).first():
+                # Return bad request if the build re-triggered is not the
+                # latest for that version
+                return HttpResponseBadRequest()
```

Comment:
What does this show to the user?

Reply:
Good catch. This shows a blank page. I'm not sure what's best, since a 400 makes more sense for this error but we probably just want to trigger our default 404?

The user will hit this scenario when a new build for the PR was triggered while they have the page opened and before they clicked on this button. So, it may be more common than I originally thought.

Reply:
I added a message "This build can't be re-triggered because it's not the latest build for this version." and return a redirect to the build listing page.